### DISCUSSION TITLE
Looking forward to kd-tree implementation..

### DISCUSSION THREAD

**Discussion by jiapei100** [oneapi-src/oneDAL#2231] [Q&A] [answered]
Hi, I passed by your documentation at [https://spec.oneapi.io/versions/latest/elements/oneDAL/source/appendix/kd_tree.html](https://spec.oneapi.io/versions/latest/elements/oneDAL/source/appendix/kd_tree.html), and it seems you've got an implementation of **kd_tree** based on **dpl/pstl** with **tbb** backend ??

Not quite sure about that....  Anyway, according to the above link, can you please provide some more info ? Thank you ..

**Comment by napetrov** [comment, marked as answer]
@jiapei100  - you can look on implementation her e https://github.com/oneapi-src/oneDAL/tree/master/cpp/oneapi/dal/algo/knn/backend
We are not using dpl/pstl. Basically there is sycl based implementation
